src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFSurface3DChartData.java

```diff
@@ -0,0 +1,168 @@
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+==================================================================== */
+
+package org.apache.poi.xddf.usermodel.chart;
+
+import java.util.Map;
+
+import org.apache.poi.util.Beta;
+import org.apache.poi.xddf.usermodel.XDDFShapeProperties;
+import org.openxmlformats.schemas.drawingml.x2006.chart.CTAxDataSource;
+import org.openxmlformats.schemas.drawingml.x2006.chart.CTBoolean;
+import org.openxmlformats.schemas.drawingml.x2006.chart.CTNumDataSource;
+import org.openxmlformats.schemas.drawingml.x2006.chart.CTSerTx;
+import org.openxmlformats.schemas.drawingml.x2006.chart.CTSurface3DChart;
+import org.openxmlformats.schemas.drawingml.x2006.chart.CTSurfaceSer;
+
+@Beta
+public class XDDFSurface3DChartData extends XDDFChartData {
+    private CTSurface3DChart chart;
+
+    public XDDFSurface3DChartData(CTSurface3DChart chart, Map<Long, XDDFChartAxis> categories,
+            Map<Long, XDDFValueAxis> values) {
+        this.chart = chart;
+        for (CTSurfaceSer series : chart.getSerList()) {
+            this.series.add(new Series(series, series.getCat(), series.getVal()));
+        }
+        defineAxes(categories, values);
+    }
+
+    private void defineAxes(Map<Long, XDDFChartAxis> categories, Map<Long, XDDFValueAxis> values) {
+        if (chart.sizeOfAxIdArray() == 0) {
+            for (Long id : categories.keySet()) {
+                chart.addNewAxId().setVal(id);
+            }
+            for (Long id : values.keySet()) {
+                chart.addNewAxId().setVal(id);
+            }
+        }
+        defineAxes(chart.getAxIdArray(), categories, values);
+    }
+    
+    public void setSeriesAxisId(XDDFSeriesAxis seriesAxis)
+    {
+            chart.addNewAxId().setVal(seriesAxis.getId());
+    }
+    
+    public CTBoolean getWireframe()
+    {
+        if(chart.isSetWireframe())
+            return chart.getWireframe();
+        else
+            return chart.addNewWireframe();
+    }
+    
+    public void setWireframe(boolean val)
+    {
+        if(chart.isSetWireframe())
+            chart.getWireframe().setVal(val);
+        else
+            chart.addNewWireframe().setVal(val);
+    }
+
+    @Override
+    public void setVaryColors(boolean varyColors) {
+       
```

Comment:
It might be worth a comment to explain the underlying `CTSurface3DChart` has no such property as `varyColors`.

Reply:
added comment.